### DISCUSSION TITLE
"No StateHolder Provided" Error in Compose Multiplatform Project with Koin and Navigation

### DISCUSSION THREAD

**Discussion by rkshnair** [Tlaster/PreCompose#217] [Q&A]
I'm encountering a runtime exception in my Compose Multiplatform project, specifically when integrating MVVM with PreCompose for navigation and Koin for dependency injection. The error occurs at the line `val navigator = rememberNavigator() `with the message "No StateHolder provided."

Here's the relevant portion of my code:

```
@Composable
fun NavHostHelper() {
    val navigator = rememberNavigator()
    NavHost(
        navigator = navigator,
        navTransition = NavTransition(),
        initialRoute = Screen.Login.route,
    ) {
        scene(
            route = Screen.Login.route,
            navTransition = NavTransition(),
        ) {
            val vm = koinViewModel(LoginViewModel::class)
            LoginScreen(vm) {
                navigator.navigate(Screen.Login.route+ "$it")
            }
        }
    }
}

```

**Comment by Lavmee** [comment]
Hi, have you wrapped your entry point as specified by the documentation?

https://github.com/Tlaster/PreCompose/blob/master/docs/setup.md

**Reply by rkshnair** [reply]
I have something like this. 

```
@Composable
fun App() {
    PreComposeApp {
       AppTheme {
             NavHostHelper()
       }
    }
}
```

**Reply by Lavmee** [reply]
Can you write which version of the precompose you are using and on which platform the crash occurs?

**Reply by rkshnair** [reply]
1.5.10 and I tried Android; Didn't get a chance to check IOS.

**Reply by Lavmee** [reply]
Can you send the full crash log?

**Reply by Lavmee** [reply]
I can't reproduce this crash. Maybe you can create a minimal project with this error?

**Reply by MerlinTHS** [reply]
I think he talked about that one: `Caused by: org.koin.core.error.NoBeanDefFoundException: No definition found for type 'moe.tlaster.precompose.stateholder.SavedStateHolder'. Check your Modules configuration and add missing type and/or qualifier!
` which occurs when I assume to get an Instance of `SavedStateHolder` in a `ViewModel` constructor which will be invoked by Koin.

**Comment by rkshnair** [comment]
i decided to go with voyager. So closing the thread for the time being.
